client/my-sites/upgrades/checkout-thank-you/business-plan-details.jsx

```diff
@@ -46,7 +46,10 @@ const BusinessPlanDetails = ( { selectedSite } ) => {
 };
 
 BusinessPlanDetails.propTypes = {
-	selectedSite: React.PropTypes.object.isRequired
+	selectedSite: React.PropTypes.oneOfType( [
+		React.PropTypes.bool,
```

Comment:
Well if you clear your local storage and reload the page `selectedSite` will be `false`. 

>  we assume the presence of a selected site when fetching the receipt in CheckoutThankYou

Do we? I don't see [any such thing](https://github.com/Automattic/wp-calypso/blob/56d4772f2954ba11322ab02ef4d04fa545254fcc/client/my-sites/upgrades/checkout-thank-you/index.jsx#L66) in the `CheckoutThankYou` component. I might be missing something so if you could point me to where that happens in our code that would be great.

> This component accesses properties of selectedSite without handling the case where selectedSite is false.

That's a problem then, good catch. If we indeed execute `localStorage.clear()` and reload the `Thank You` page it will indeed display with missing data and `undefined` values. I'm going to address that in a different pull request though.